src/padding.rs

```diff
@@ -1,7 +1,87 @@
+use std::fmt;
+
+use digest::{Digest, DynDigest};
+use rand::RngCore;
+
+use crate::hash::Hash;
+
 /// Available padding schemes.
-#[derive(Debug, Clone, Copy)]
 pub enum PaddingScheme {
-    PKCS1v15,
-    OAEP,
-    PSS,
+    /// Encryption and Decryption using PKCS1v15 padding.
+    PKCS1v15Encrypt,
+    /// Sign and Verify using PKCS1v15 padding.
+    PKCS1v15Sign { hash: Option<Hash> },
```

Comment:
The `hash: None` case appears to still be necessary for unpadded signatures (at least, that's what the test case in `src/pkcs1v15.rs` leads me to believe).

Reply:
Yes, which is why it is still an `Option`, not sure what you mean.

Reply:
I was just noting this because it wasn't immediately obvious during review that the `Option<Hash>` was necessary besides enabling the prior encryption case to specify `None`.